internal/common/utils/misc.go

```diff
@@ -109,3 +109,21 @@ func LoadSchemas(cfg *viper.Viper, schemaNames []string) (schemas []*jsonschema.
 	}
 	return
 }
+
+type apiVersionCtxKeyType string
+
+const apiCtxKey apiVersionCtxKeyType = "api_version"
+
+func WithApiVersion(ctx context.Context, value string) context.Context {
+	return context.WithValue(ctx, apiCtxKey, value)
+}
+
+func GetApiVersion(ctx context.Context) string {
+	result, ok := ctx.Value(apiCtxKey).(string)
+
+	if !ok {
+		return "unknown"
+	}
+
+	return result
+}
```

Comment:
What do you think about moving this code into internal/api/utils to keep this logic closer to the api area of the code where its being used?